Warns maybe-uninitialized on g++-6

A quick one:
```shellsession
$ cmake -DCMAKE_CXX_COMPILER=g++-6 ..
```
causes this:
```shellsession
In file included from /opt/jpmag/proj/extern/fast_float/include/fast_float/ascii_number.h:9:0,
                 from /opt/jpmag/proj/extern/fast_float/include/fast_float/parse_number.h:3,
                 from /opt/jpmag/proj/extern/fast_float/include/fast_float/fast_float.h:44,
                 from /opt/jpmag/proj/extern/fast_float/tests/random_string.cpp:1:
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h: In function ‘fast_float::adjusted_mantissa fast_float::parse_long_mantissa(const char*, const char*) [with binary = fast_float::binary_format<double>]’:
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h:134:35: error: ‘answer.fast_float::adjusted_mantissa::mantissa’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     return mantissa == o.mantissa && power2 == o.power2;
            ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /opt/jpmag/proj/extern/fast_float/include/fast_float/parse_number.h:4:0,
                 from /opt/jpmag/proj/extern/fast_float/include/fast_float/fast_float.h:44,
                 from /opt/jpmag/proj/extern/fast_float/tests/random_string.cpp:1:
/opt/jpmag/proj/extern/fast_float/include/fast_float/decimal_to_binary.h:73:21: note: ‘answer.fast_float::adjusted_mantissa::mantissa’ was declared here
   adjusted_mantissa answer;
                     ^~~~~~
make[1]: *** [CMakeFiles/Makefile2:216: tests/CMakeFiles/long_exhaustive32_64.dir/all] Error 2
```

There is also this:
```shellsession
opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h: In constructor ‘fast_float::decimal::decimal()’:
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h:144:3: error: ‘fast_float::decimal::num_digits’ should be initialized in the member initialization list [-Werror=effc++]
   decimal() = default;
   ^~~~~~~
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h:144:3: error: ‘fast_float::decimal::decimal_point’ should be initialized in the member initialization list [-Werror=effc++]
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h:144:3: error: ‘fast_float::decimal::negative’ should be initialized in the member initialization list [-Werror=effc++]
/opt/jpmag/proj/extern/fast_float/include/fast_float/float_common.h:144:3: error: ‘fast_float::decimal::truncated’ should be initialized in the member initialization list [-Werror=effc++]
In file included from /opt/jpmag/proj/extern/fast_float/include/fast_float/parse_number.h:3:0,
                 from /opt/jpmag/proj/extern/fast_float/include/fast_float/fast_float.h:44,
                 from /opt/jpmag/proj/extern/fast_float/tests/long_exhaustive32_64.cpp:1:
/opt/jpmag/proj/extern/fast_float/include/fast_float/ascii_number.h: In function ‘fast_float::decimal fast_float::parse_decimal(const char*, const char*)’:
/opt/jpmag/proj/extern/fast_float/include/fast_float/ascii_number.h:169:11: note: synthesized method ‘fast_float::decimal::decimal()’ first required here 
   decimal answer;
           ^~~~~~
```

--- a/.github/workflows/msys2.yml
+++ b/.github/workflows/msys2.yml
@@ -35,10 +35,11 @@ jobs:
           update: true
           msystem: ${{ matrix.msystem }}
           install: ${{ matrix.install }}
-      - name: Build and Test
-        run: |
-          mkdir build
-          cd build
-          cmake -DCMAKE_BUILD_TYPE=${{ matrix.type }}  -DFASTFLOAT_TEST=ON ..
-          cmake --build . --verbose
-          ctest  --output-on-failure -R basictest
+      - name: Prepare build dir
+        run: mkdir build
+      - name: Configure
+        run: cd build && cmake -DCMAKE_BUILD_TYPE=${{ matrix.type }} -DFASTFLOAT_TEST=ON ..
+      - name: Build
+        run: cmake --build build
+      - name: Run basic tests
+        run: cd build && ctest --output-on-failure -R basictest

--- a/.github/workflows/ubuntu18.yml
+++ b/.github/workflows/ubuntu18.yml
@@ -1,20 +1,32 @@
-name: Ubuntu 18.04 CI (GCC 7)
+name: Ubuntu 18.04 CI (GCC 7, 6, 5)
 
 on: [push, pull_request]
 
 jobs:
   ubuntu-build:
     runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - {cxx: -DCMAKE_CXX_COMPILER=g++-5}
+          - {cxx: -DCMAKE_CXX_COMPILER=g++-6}
+          - {cxx: } # default compiler 7
     steps:
       - uses: actions/checkout@v2
       - name: Setup cmake
         uses: jwlawson/actions-setup-cmake@v1.0
         with:
           cmake-version: '3.9.x'
-      - name: Use cmake
+      - name: Install older compilers
         run: |
-          mkdir build &&
-          cd build &&
-          cmake  -DFASTFLOAT_TEST=ON  .. &&
-          cmake --build . &&
-          ctest --output-on-failure -R basictest
+          sudo -E apt-get update
+          sudo -E apt-get install -y --force-yes g++-5 g++-6
+      - name: Prepare build dir
+        run: mkdir build
+      - name: Configure
+        run: cd build && cmake ${{matrix.cxx}} -DFASTFLOAT_TEST=ON ..
+      - name: Build
+        run: cmake --build build
+      - name: Run basic tests
+        run: cd build && ctest --output-on-failure -R basictest

--- a/.github/workflows/ubuntu20.yml
+++ b/.github/workflows/ubuntu20.yml
@@ -1,20 +1,31 @@
-name: Ubuntu 20.04 CI (GCC 9)
+name: Ubuntu 20.04 CI (GCC 9, 8)
 
 on: [push, pull_request]
 
 jobs:
   ubuntu-build:
     runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - {cxx: -DCMAKE_CXX_COMPILER=g++-8}
+          - {cxx: } # default compiler 9
     steps:
       - uses: actions/checkout@v2
       - name: Setup cmake
         uses: jwlawson/actions-setup-cmake@v1.0
         with:
           cmake-version: '3.9.x'
-      - name: Use cmake
+      - name: install older compilers
         run: |
-          mkdir build &&
-          cd build &&
-          cmake  -DFASTFLOAT_TEST=ON  .. &&
-          cmake  --build .   &&
-          ctest --output-on-failure  -R basictest
+          sudo -E apt-get update
+          sudo -E apt-get install -y --force-yes g++-8 g++-7
+      - name: Prepare build dir
+        run: mkdir build
+      - name: Configure
+        run: cd build && cmake ${{matrix.cxx}} -DFASTFLOAT_TEST=ON ..
+      - name: Build
+        run: cmake --build build
+      - name: Run basic tests
+        run: cd build && ctest --output-on-failure -R basictest

